### ISSUE TITLE
Improve node pool experience

### ISSUE THREAD

**Issue by ryanfaircloth** [humio/humio-operator#666]
When creating a node pool a substantial portion of the work is copy and past making the CR difficult to sustain.

* When using node pools the "root spec" should be a template for the pools and not create a pool itself
* Each pool (ALL, ui, ingest) should be defined in node pools
* Information from the root spec should be the base for each pool with the individual pool choices taking precidence

**Comment by jswoods** [comment]
I agree there can be some improvement here. We are working to enable a config that will allow us to define only node pools and nothing at the "base" spec. The goal would be to eventually remove these types of configs from the base part of the spec.

In order to align with that plan and not add more confusion about inheritance, I would suggest we add a new field to the base spec, along the lines of `defaultNodePoolAttributes` or `commonNodePoolAttributes`, which then are inherited by any of the node pools that are defined in the node pool list.

**Comment by ryanfaircloth** [comment]
I like this solution it's easy to understand.
